backend/src/main/java/gov/cdc/usds/simplereport/service/PatientBulkUploadServiceAsync.java

```diff
@@ -0,0 +1,149 @@
+package gov.cdc.usds.simplereport.service;
+
+import static gov.cdc.usds.simplereport.api.Translators.parsePersonRole;
+import static gov.cdc.usds.simplereport.api.Translators.parsePhoneType;
+import static gov.cdc.usds.simplereport.api.Translators.parseUserShortDate;
+import static gov.cdc.usds.simplereport.api.Translators.parseYesNo;
+import static gov.cdc.usds.simplereport.validators.CsvValidatorUtils.convertEthnicityToDatabaseValue;
+import static gov.cdc.usds.simplereport.validators.CsvValidatorUtils.convertRaceToDatabaseValue;
+import static gov.cdc.usds.simplereport.validators.CsvValidatorUtils.convertSexToDatabaseValue;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import gov.cdc.usds.simplereport.api.model.filerow.PatientUploadRow;
+import gov.cdc.usds.simplereport.config.AuthorizationConfiguration;
+import gov.cdc.usds.simplereport.db.model.Facility;
+import gov.cdc.usds.simplereport.db.model.Organization;
+import gov.cdc.usds.simplereport.db.model.Person;
+import gov.cdc.usds.simplereport.db.model.PhoneNumber;
+import gov.cdc.usds.simplereport.db.model.auxiliary.StreetAddress;
+import gov.cdc.usds.simplereport.validators.CsvValidatorUtils;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class PatientBulkUploadServiceAsync {
+
+  private final PersonService _personService;
+  private final AddressValidationService _addressValidationService;
+  private final OrganizationService _organizationService;
+
+  @Async
+  @Transactional
+  @AuthorizationConfiguration.RequirePermissionCreatePatientAtFacility
+  public CompletableFuture<Set<Person>> savePatients(byte[] content, UUID facilityId) {
+    Organization currentOrganization = _organizationService.getCurrentOrganization();
+
+    // Patients do not need to be assigned to a facility, but if an id is given it must be valid
+    Optional<Facility> assignedFacility =
+        Optional.ofNullable(facilityId).map(_organizationService::getFacilityInCurrentOrg);
+
+    Set<Person> patientsList = new HashSet<>();
```

Comment:
this Set is still error prune, and susceptible to and out-of-memory error given a big enough file, since it will grow to the max number of the uploaded file.